### PR TITLE
refactor(generate-readme): switch to cargo-rdme

### PR DESCRIPTION
**Pull request by jost-s**
### Summary
`cargo-readme` has been used to generate crate level READMEs from doc comments. The resulting READMEs contained invalid links. Symbol resolution from intra-doc links didn't work.

Now switched to another tool - `cargo-rdme` (note the subtle difference here!). It also doesn't resolve links properly, but is a slightly more actively maintained tool and plans to implement this feature. Instead of resolving the links, they are simply stripped for now. At least that produces a valid README.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [x] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Comment by jost-s** [comment]
It works on CI https://github.com/holochain/holochain/actions/runs/4020425563/jobs/6908345157#step:12:96

File: crates/hdi/README.md [hunk 1]
```diff
@@ -9,92 +9,7 @@
...
-            "No Bs allowed in this app".to_string(),
-        )),
-    },
-    _ => Ok(ValidateCallbackResult::Valid),
-};
-```
-See an example of the `validate` callback in an integrity zome in the WASM workspace:
-<https://github.com/holochain/holochain/blob/develop/crates/test_utils/wasm/wasm_workspace/validate/src/integrity.rs>.
-Many more validation examples can be browsed in that very workspace.
+<!-- cargo-rdme -->
```
Comment (neonphog) [inline crates/hdi/README.md:12]: It doesn't appear to be working.

**Review by neonphog — CHANGES_REQUESTED** [review]
not sure if I'm misunderstanding something, but I don't think it's acceptable for the readmes to not be viewable on github.

File: crates/hdi/README.md (hunk 1 above)
Comment (jost-s) [reply crates/hdi/README.md]: It's working, just that the files are not yet generated. You can run the script locally to test it.
I've added the generated files anyway.

**Comment by jost-s** [comment]
> not sure if I'm misunderstanding something, but I don't think it's acceptable for the readmes to not be viewable on github.

They are viewable just like before, but the intra-doc links like `[AdminResponse::InstallApp](crate::AdminResponse::InstallApp)` are stripped on Github and result in plain text `AdminResponse::InstallApp`.

Previously the intra-doc links were copied across to Github README as is and wouldn't work.

**Comment by jost-s** [comment]
You can check the README's diff now for comparison.

**Review by neonphog — APPROVED** [review]
: / - I guess, I'm not a huge fan of the actual generation happening after the fact. I'd rather the ci step be just a verification that everything was done correctly, so that we could review what will actually be posted in the PR review.

But, I'll approve this for now, and we can consider that at a later stage : )

**Comment by jost-s** [comment]
> : / - I guess, I'm not a huge fan of the actual generation happening after the fact. I'd rather the ci step be just a verification that everything was done correctly, so that we could review what will actually be posted in the PR review.
> 
> But, I'll approve this for now, and we can consider that at a later stage : )

I hear you. This is a solution so that you don't need to consider the crate READMEs when you're editing the doc comments, and it's just swapping the generator.

I'm totally up for coming up with better ways of doing that.

**Pull request merged**